frontend/app/routes/protected/renew/$id/verify-email.tsx

```diff
@@ -69,32 +69,20 @@ export async function action({ context: { appContainer, session }, params, reque
   await securityHandler.validateAuthSession({ request, session });
   securityHandler.validateCsrfToken({ formData, session });
 
-  const idToken: IdToken = session.get('idToken');
   const state = loadProtectedRenewState({ params, request, session });
   const t = await getFixedT(request, handle.i18nNamespaces);
   const { ENGLISH_LANGUAGE_CODE } = appContainer.get(TYPES.configs.ServerConfig);
+  const { ENABLED_FEATURES } = appContainer.get(TYPES.configs.ClientConfig);
+  const demographicSurveyEnabled = ENABLED_FEATURES.includes('demographic-survey');
 
   // Fetch verification code service
   const verificationCodeService = appContainer.get(TYPES.domain.services.VerificationCodeService);
 
   const formAction = z.nativeEnum(FORM_ACTION).parse(formData.get('_action'));
 
-  if (formAction === FORM_ACTION.back) {
-    saveProtectedRenewState({
-      params,
-      request,
-      session,
-      state: {
-        emailVerified: state.emailPreviouslyValidated,
-      },
-    });
-
-    return redirect(getPathById('protected/renew/$id/confirm-email', params));
-  }
-
   if (formAction === FORM_ACTION.request) {
     // Create a new verification code and store the code in session
-    const verificationCode = verificationCodeService.createVerificationCode(idToken.sub);
+    const verificationCode = verificationCodeService.createVerificationCode('anonymous');
```

Comment:
The `userId` should be left as is.

```suggestion
    const verificationCode = verificationCodeService.createVerificationCode(idToken.sub);
```

(there's a few occurrences of this change throughout the files in this PR)